### DISCUSSION TITLE
Preventing files from being cleared on save with `biome check` using `--stdin-file-path`

### DISCUSSION THREAD

**Discussion by stewarthsoj** [biomejs/biome#522] [Q&A] [answered]
I'm currently using the `biome check` command to format code and organize imports, but when there is nothing to fix the output is empty. The problem with this is when I want to run this command on save with editors like [Zed](https://zed.dev) the contents of the file is cleared.

As an example, here is how I set up `biome check` to run on save in Zed:

```
{
  "language_overrides": {
    "TSX": {
      "format_on_save": {
        "external": {
          "command": "bunx",
          "arguments": [
            "biome",
            "check",
            "--apply-unsafe",
            "--stdin-file-path={buffer_path}"
          ]
        }
      }
    }
  }
}
```

Here is an example TSX file:

```
import { a, b } from 'dep';

export default function Component() {
  return <pre>Testing</pre>;
}
```

If I save this file in Zed in the current state, the file is cleared. But, if I remove, say, the semi-colon, or change the order of the imports (i.e. `b` is first), the file is formatted correctly and the imports are organized.

In summary, I'm keen to know if there is a workaround for this i.e. a way to abort piping empty inputs to stdout or always passing the file contents.

**My Biome Config**

```
{
  "$schema": "https://biomejs.dev/schemas/1.2.2/schema.json"
}
```

**References**

- https://docs.zed.dev/languages/javascript
- https://discord.com/channels/1132231889290285117/1155222438007812249/1162555949068861550

**Comment by ematipico** [comment, marked as answer]
That's probably a bug 🐛

There's no current workaround, so opening a PR and fixing the issue is the best course of action.

And I think the fix is to return the original output regardless of the outcome of the command.

**Comment by ematipico** [comment]
Resolved in https://github.com/biomejs/biome/pull/543. The next release will have this fix

**Reply by stewarthsoj** [reply]
Amazing!
